### ISSUE TITLE
Add support for Discriminator Object

### ISSUE THREAD

**Issue by anton-petrov** [koxudaxi/datamodel-code-generator#957]
**Is your feature request related to a problem? Please describe.**
We need to have correct support of discriminator object type. At the moment the behavior differs from the required one, instead of adding the discirminator attribute to the field, a dictionary with links to schemas is created:
```python
# wrong autogenerated code
Field(
    ...,
    discriminator={
                "propertyName": "type",
                "mapping": {
                    "type1": "#/components/schemas/Schema1",
                    "type2": "#/components/schemas/Schema2",
                    "type3": "#/components/schemas/Schema3",
                },
            },
)
```

**Describe the solution you'd like**
As an example, take a look at a piece of the api specification and the resulting code:

_OpenAPI_
```yaml
    Schema:
      oneOf:
        - $ref: "#/components/schemas/Schema1"
        - $ref: "#/components/schemas/Schema2"
        - $ref: "#/components/schemas/Schema3"
      discriminator:
        propertyName: type
        mapping:
          type1: "#/components/schemas/Schema1"
          type2: "#/components/schemas/Schema2"
          type3: "#/components/schemas/Schema3"
```

_Generated pydantic model_
```python
class OptimizationSimulationParametersSchema(BaseModel):
    __root__: Union[
        Schema1,
        Schema2,
        Schema3,
    ] = Field(..., discriminator="type")
```

**Additional info**
Link to specification https://spec.openapis.org/oas/v3.0.3#discriminator-object

**Comment by koxudaxi** [comment]
@anton-petrov 
 I have released a new version 0.17.0.
The version support OpenAPI Discriminator object.
Thank you very much!!

**Comment by anton-petrov** [comment]
@koxudaxi Great work! Thank you very much for such amazing project!
